temporal-sdk/src/main/java/io/temporal/internal/client/RootWorkflowClientInvoker.java

```diff
@@ -27,28 +27,52 @@
 import io.temporal.client.WorkflowClientOptions;
 import io.temporal.common.interceptors.WorkflowClientCallsInterceptor;
 import io.temporal.internal.client.external.GenericWorkflowClient;
+import io.temporal.worker.WorkflowTaskDispatchHandle;
 import java.lang.reflect.Type;
 import java.util.*;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.TimeoutException;
+import javax.annotation.Nullable;
 
 public class RootWorkflowClientInvoker implements WorkflowClientCallsInterceptor {
   private final GenericWorkflowClient genericClient;
   private final WorkflowClientOptions clientOptions;
+  private final EagerWorkflowTaskDispatcher eagerWorkflowTaskDispatcher;
   private final WorkflowClientRequestFactory requestsHelper;
 
   public RootWorkflowClientInvoker(
-      GenericWorkflowClient genericClient, WorkflowClientOptions clientOptions) {
+      GenericWorkflowClient genericClient,
+      WorkflowClientOptions clientOptions,
+      WorkerFactoryRegistry workerFactoryRegistry) {
     this.genericClient = genericClient;
     this.clientOptions = clientOptions;
+    this.eagerWorkflowTaskDispatcher = new EagerWorkflowTaskDispatcher(workerFactoryRegistry);
     this.requestsHelper = new WorkflowClientRequestFactory(clientOptions);
   }
 
   @Override
   public WorkflowStartOutput start(WorkflowStartInput input) {
-    StartWorkflowExecutionRequest request =
-        requestsHelper.newStartWorkflowExecutionRequest(input).build();
-    return new WorkflowStartOutput(genericClient.start(request));
+    try (WorkflowTaskDispatchHandle eagerDispatchHandle = obtainDispatchHandle(input)) {
```

Comment:
```suggestion
    // Handle is null if user disabled via options or no worker slot available
    try (WorkflowTaskDispatchHandle eagerDispatchHandle = obtainDispatchHandle(input)) {
```

Was a bit unclear to me before I started digging into other code more